code.js

```diff
@@ -1,3 +1,46 @@
-function dijkstra(graph, sourceNode) {
-    return [];
+// - initialize the dist to each vertex to $\infty$, source to 0
+// - while there are unmarked vertices left in the graph
+//     - select the unmarked vertex $v$ with the lowest dist
+//     - mark $v$ with distance dist
+//     - for each edge $(v,w)$
+//         - dist($w$) = min $\left(\textrm{dist}(w), \textrm{dist}(v) + \textrm{weight of }(v, w)\right)$
+
+/**
+ * @param {number[][]} graph
+ * @param {number} start
+ */
+function dijkstra(graph, start) {
+  // initialize the dist to each vertex to $\infty$, source to 0
+  let distances = new Array(graph.length).fill(Infinity);
+  let visited = new Set();
+  distances[start] = 0;
+
+  // while there are unmarked vertices left in the graph
+  while (visited.size < graph.length) {
+    //  select the unmarked vertex $v$ with the lowest dist
+    let [minDistance, closestNode] = distances.reduce(
```

Comment:
Would this step here not make it quadratic in the number of vertices?

Reply:
Yes, you are correct. I fixed my analysis for this section and thus the overall algorithm.